aac_codec_registration.src.html

```diff
@@ -0,0 +1,178 @@
+================================================================
+
+If {{AudioDecoderConfig.description}} is present, it is assumed to a
+`AudioSpecificConfig` as defined in [[iso14496-3]] section 1.6.2.1, Table 1.15,
+and the bitstream is assumed to be in {{AacBitstreamFormat/aac}}.
+
+If the {{AudioDecoderConfig.description}} is not present, ht ebitstream is
+assumed to be in {{AacBitstreamFormat/adts}} format.
+
+The {{AudioDecoderConfig.sampleRate}} and {{AudioDecoderConfig.channelCount}}
+members are ignored.
+
+AudioEncoderConfig extensions {#audioencoderconfig-extensions}
+==============================================================
+
+<pre class='idl'>
+<xmp>
+partial dictionary AudioEncoderConfig {
+  AacEncoderConfig aac;
+};
+</xmp>
+</pre>
+
+<dl>
+  <dt><dfn dict-member for=AudioEncoderConfig>aac</dfn></dt>
+  <dd>
+    Contains codec specific configuration options for the AAC codec.
+  </dd>
+</dl>
+
+AacEncoderConfig {#aac-encoder-config}
+--------------------------------------
+
+<pre class='idl'>
+<xmp>
+dictionary AacEncoderConfig {
+  AacBitstreamFormat format = "aac";
+};
+</xmp>
+</pre>
+
+<dl>
+  <dt><dfn dict-member for=AacEncoderConfig>format</dfn></dt>
+  <dd>
+    Configures the format of output {{EncodedAudioChunk}}s. See
+    {{AacBitstreamFormat}}.
+  </dd>
+</dl>
+
+AacBitstreamFormat {#aac-bitstream-format}
+------------------------------------------
+<pre class='idl'>
+<xmp>
+enum AacBitstreamFormat {
+  "aac",
+  "adts",
+};
+</xmp>
+</pre>
+
+The {{AacBitstreamFormat}} determines the location of the metadata necessary to decode the encoded audio stream.
+
+<dl>
+  <dt><dfn enum-value for=AacBitstreamFormat>aac</dfn></dt>
+  <dd>
+    The metadata of the encoded audio stream are provided at configuration via
+    {{AudioDecoderConfig.description}}.
+  </dd>
+  <dt><dfn enum-value for=AacBitstreamFormat>adts</dfn></dt>
+  <dd>
+    The metadata of the encoded audio stream are provided in each ADTS frame,
+    and therefore no {{AudioDecoderConfig.description}} is necessary.
+</dl>
+
+
+EncodedAudioChunk data {#encodedaudiochunk-data}
+================================================
+
+If the bitstream is in {{AacBitstreamFormat/adts}} format, the
+[=EncodedAudioChunk/[[internal data]]=] of {{EncodedAudioChunk}}s are expected
+to be an ADTS frame, as described in section 1.A.3.2 of [[iso14496-3]].
+
+If the bitstream is in {{AacBitstreamFormat/aac}} format, the
+[=EncodedAudioChunk/[[internal data]]=] of {{EncodedAudioChunk}}s are expected
+to be a raw AAC frame (syntax element `raw_data_block()`), as described in
+section 4.4.2.1 of [[iso14496-3]].
+
+
+EncodedAudioChunk type {#encodedaudiochunk-type}
+================================================
+
+The [=EncodedAudioChunk/[[type]]=] for an {{EncodedAudioChunk}} containing
+AAC is always "[=EncodedAudioChunkType/key=]".
+
+NOTE: Once the initialization has succeeded, any AAC packet can be decoded at
```

Comment:
What's the history of this ominous comment? I saw it on some other registrations too.

Reply:
It's just to warn authors that no error will be reported, ever, if the packet are shuffled around, but the audio will be incorrect, since the decoders are stateful, unlike with some video codecs/implementation of those codecs, that have various types of frames  that need to be fed in order.